filebeat/input/filestream/internal/input-logfile/manager_test.go

```diff
@@ -275,6 +277,104 @@ paths:
 		}
 	})
 
+	t.Run("failed input has its ID removed from the IDs list", func(t *testing.T) {
+		storeReg := statestore.NewRegistry(storetest.NewMemoryStoreBackend())
+		testStore, err := storeReg.Get("test")
+		require.NoError(t, err)
+
+		log, _ := newBufferLogger()
+
+		cim := &InputManager{
+			Logger:     log,
+			StateStore: testStateStore{Store: testStore},
+			Configure: func(cfg *config.C) (Prospector, Harvester, error) {
+				var wg sync.WaitGroup
+
+				settings := struct {
+					ID    string   `config:"id"`
+					Paths []string `config:"paths"`
+				}{}
+
+				if err := cfg.Unpack(&settings); err != nil {
+					return nil, nil, err
+				}
+
+				for _, path := range settings.Paths {
+					if strings.Contains(path, "**/**") {
+						return nil, nil, errors.New("double ** is not allowed in a glob")
+					}
+				}
+
+				return &noopProspector{}, &mockHarvester{onRun: correctOnRun, wg: &wg}, nil
+			}}
+		invalidCfg := config.MustNewConfigFrom(`
+type: filestream
+id: t-wing
+paths:
+  - "/**/**/foo" # double ** is invalid for Filestream
+`)
+
+		// Create a valid config with the same ID
+		validCfg := config.MustNewConfigFrom(`
+type: filestream
+id: t-wing
+paths:
+  - /var/log/bar
+`)
+
+		// Happy path, if an input fails to start, it's ID is removed from cim.ids list and
+		// can be re-used.
+		t.Run("happy path", func(t *testing.T) {
+			// Attempt to create the first input with the invalid configuration
+			_, err = cim.Create(invalidCfg)
+			require.Error(
+				t,
+				err,
+				"'/**/**' is not supported, input creation must fail")
+
+			require.Len(t, cim.ids, 0, "no ID must be present in cim.ids")
+			// Attempt to create the second input with the valid configuration
+			_, err = cim.Create(validCfg)
+			require.NoError(
+				t,
+				err,
+				"The same ID can be re-used after an input fails to start")
+			require.EqualValues(
+				t,
+				map[string]struct{}{"t-wing": {}},
+				cim.ids,
+				"only 't-wing' must be present in cim.ids")
+
+			// "Stop" the input to have the manager in a consistent state
+			// using the same flow as an actually running input would
+			cim.StopInput("t-wing")
+		})
+
+		// Failure scenario: an input with ID X is already running, then an invalid
+		// input configuration with the same ID is used while
+		// 'allow_deprecated_id_duplication: true'. In this case, the invalid input
+		// must fail to start, but the ID from the valid one must stay in cim.ids
+		t.Run("running input with the same ID as invalid one", func(t *testing.T) {
+			// Attempt to create the second input with the valid configuration
```

Comment:
[Nit]
'second' here make it seem like there is already an input running with this ID
```suggestion
			// Attempt to create the input with the valid configuration
```